### PR TITLE
4890 plotting NaNs

### PR DESCRIPTION
**Pull request by sluhtoja**

### CODE REVIEW

**Comment by sluhtoja** [comment]
kick jenkins

**Comment by sluhtoja** [comment]
kick jenkins

**Comment by scottdraves** [comment]
thx, the way the dialog works looks good and thank you for moving the template to its own file.
the plot still seems broken though, i am getting this:
screen shot 2016-11-24 at 2 28 39 pm

**Comment by scottdraves** [comment]
and if i do a plot with valid data, i still get the popup:
```
r = 10
xs = (0..r).collect { x-> (x-r/2)/r }
new Plot() << new Points(x:xs, y: xs.collect { x -> Math.sqrt(x*x) })
```
in fact if you just open the plotting tutorials you will see that the popup appears for many normal plots.

**Comment by scottdraves** [comment]
the data appears to be transformed into pixel coordinates already.
it should report the error as the data is originally passed in.
it should check the data the first time it scans it.
for example, in this code:
```
r = 10
xs = (0..r).collect { x-> (x-r/2)/r }
new Plot() << new Points(x:xs, y: (0..9)+"asdf")
```
it should report the bad row as having a string "asdf"

**Comment by scottdraves** [comment]
much better :)

File: core/src/main/web/outputdisplay/bko-plot/plotitems/std/plotpoint.js [hunk 1]
```diff
@@ -114,7 +114,7 @@
       if (this.vlength === 0) {
         this.clear(scope);
       } else {
-        this.draw(scope);
+        this.draw(scope)
```
Comment (scottdraves) [inline core/src/main/web/outputdisplay/bko-plot/plotitems/std/plotpoint.js:117]: keep ;

File: core/src/main/web/outputdisplay/bko-plot/plotitems/std/plotpoint.js [hunk 2]
```diff
@@ -316,11 +316,13 @@
           default:  // rect
             shapesvg.selectAll(tag)
               .data(eleprops, function(d) { return d.id; })
-              .attr("x", function(d) { return d.x; })
-              .attr("y", function(d) { return d.y; })
+              .attr("x", function(d) {return d.x;})
+              .attr("y", function(d) {return d.y;})
               .attr("width", function(d) { return d.w; })
               .attr("height", function(d) { return d.h; });
+
```
Comment (scottdraves) [inline core/src/main/web/outputdisplay/bko-plot/plotitems/std/plotpoint.js:323]: please do not add blank lines

**Comment by scottdraves** [comment]
i tried the category plot tutorial, got this error:
```
33beakerVendor.js:63042 TypeError: Cannot read property 'length' of undefined
    at ChildScope.scope.standardizeData (http://127.0.0.1:8801/39c0ebd/beaker/app/dist/beakerOutputDisplay.js:15985:54)
```
no plots appear.

**Comment by scottdraves** [comment]
2nd example main charting tutorial by showing empty popup:
screen shot 2016-12-07 at 9 29 00 am

**Comment by sluhtoja** [comment]
Actually there are invalid points but without running the cell they are not shown in the pop-up...

image

**Comment by sluhtoja** [comment]
kick jenkins

**Comment by sluhtoja** [comment]
kick jenkins

**Comment by sluhtoja** [comment]
kick jenkins

**Comment by scottdraves** [comment]
great thx that's better, but it reveals another problem, which is if there are too many rows of bad data, then the dialog overflows.  if there are so many rows that the dialog is taller than the underlying plot, then a scrollbar should appear in the dialog.  and it should show no more than 1000 rows of bad data, after that it should just stop adding them.

screen shot 2016-12-08 at 11 34 34 am

**Comment by scottdraves** [comment]
kick jenkins

**Comment by EfimovVladimir** [comment]
kick jenkins

**Pull request merged**